File: backend/app/schemas/transcript.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class TranscriptSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    video_id: str
    video_url: str
    video_title: str
    channel_name: str
    thumbnail_url: str | None = None
    publish_date: datetime
    created_at: datetime
    text_chunks: list[str]
    timestamp_chunks: list[str]


class TranscriptCreateRequest(BaseModel):
    video_url: str


class TranscriptCreateResponse(BaseModel):
    transcript: TranscriptSchema

File: frontend/types/transcript.ts
import type { LucideIcon } from "lucide-react";

export type Transcript = {
    id: number;
    videoId: string;
    videoUrl: string;
    videoTitle: string;
    channelName: string;
    thumbnailUrl: string | null;
    publishDate: string;
    createdAt: string;
    textChunks: string[];
    timestampChunks: string[];
};

export type Filetype = "txt";

export type TranscriptOptions = {
    [key: string]: boolean | Filetype;
    filetype: Filetype;
    showVideoId: boolean;
    showTitle: boolean;
    showChannelName: boolean;
    showPublishDate: boolean;
    showTimestamps: boolean;
};

export type FiletypeData = {
    type: Filetype;
    title: string;
    desc: string;
    icon: LucideIcon;
};
